### DISCUSSION TITLE
How do you generate FHIR4 patient records with ICD-10 diagnosis codes instead of SNOMED codes?

### DISCUSSION THREAD

**Discussion by telljmc** [synthetichealth/synthea#1533] [Q&A] [answered]
I have been attempting to generate FHIR records with ICD-10 codes instead of SNOMED codes for a Medicare patient population. Is there any documentation outlining how to map the SNOMED codes generated by Synthea to ICD-10 codes for exported FHIR records?  The closest I have been able to get are the comments interspersed throughout several code modules.  Other than that I have been unable to find much.   At this point, I'd like to verify my understanding and ask for input from anyone who can provide further guidance.

My findings so far:

1. Code Mapping Infrastructure:
- Synthea has a built-in code mapping system (CodeMapper class)
- Configuration is handled via synthea.properties (exporter.code_map.icd10-cm)???
- Mapping files must be in JSON format with a specific structure as documented in CodeMapper.java.  Here is an example of the JSON mapping file, src/main/resources/export/condition_code_map.json, I created for a single SNOMED code. 
```
{
  "233678006": [
    {
      "code": "J45.0",
      "description": "Predominantly allergic asthma",
      "weight": 1.0
    }
  ]
}
```

2. Current Implementation Examples:
- BB2RIFExporter appears to successfully use code mapping for Medicare data
- BB2RIFExporterTest shows an implementation using a MockMapper
- The mapping system appears to work for BB2RIF exports but isn't currently integrated with FHIR exports

3. Key Questions:
- Is it possible to export FHIR patients from Synthea with ICD-10 codes?
- Is FhirR4.java the correct place to implement ICD-10 mapping for FHIR exports?
- Should I follow the BB2RIF pattern of using CodeMapper, or is there a different recommended approach for FHIR?
- Are there existing mapping files I should be using?  I created my own SNOMED-to-ICD10 mapping file, but are there recommended best practices or a reference mapping file available?

I have not been able to generate a FHIR patient record with ICD-10 codes no matter how I have set parameters or the mapping files created.  I would appreciate any information you can provide that would let me know whether what I am doing is possible.  If it is possible, then I would appreciate if you could provide any of the following:
1. A brief outline of the steps required to create FHIR patient records with ICD-10 diagnosis codes.
2. a pointer to documentation that describes how to export patient FHIR records with code mappings other than SNOMED.  
3. existing code or mapping files of which should be aware?  I am assuming that mappings for codes such as CPT codes will require an AMA license and I will need to map the CPT codes to SNOMED codes?  But what about HCPCS, DME, etc.

**Comment by hadleynet** [comment, marked as answer]
What you are looking to do should work with the latest master branch code (since this PR was merged: https://github.com/synthetichealth/synthea/pull/1521). A couple of things to check:

1. In `synthea.properties` uncomment the line `exporter.code_map.icd10-cm=export/anti_amyloid_code_map.json` and change the mapping file name to whatever you are using.
2. Make sure the mapping file is in src/main/resources/export - the `export` path in the `synthea.properties` file is relative to `src/main/resources`

You can find a couple of sample code mapping files in `src/test/resources/export`.

**Reply by telljmc** [reply]
Marc, thank you for replying and for pointing out the sample mapping files.  I verified that my mapping file, icd10_code_map.json was identical in format to the samples you suggested I look at.  I had previously set up my `synthea.properties` file to point to the mapping file in the export folder as you recommended above by inserting the following two lines at the end of the file:

```
# enable mapping of Synthea-native code systems to others
exporter.code_map.icd10-cm = export/icd10_code_map.json
```

I decided to take one more step and reinstall Synthea since I had made several other changes to the parameters file and had inserted several `println` statements in some of the Exporter modules to see the flow.  I then added the two lines above to the default version of the properties file and copied my mapping file, `export/icd10_code_map.json` from the previous install to the new install.

I then ran Synthea using my standard parameters of `./run_synthea -p 1 -s 12345 Georgia` and it ran successfully.  I suspect a setting in my previous parameters file was likely the source of my problems.

One observation and two questions.  I assumed Synthea would replace the SNOMED code with the corresponding ICD10 code.  However, both are present in the output file (see below).  
1. Is this the expected behavior or is there a way to only display the ICD-10 code?  The FHIR output for the only condition for which I created a SNOMED to ICD10 map, 233678006 --> J45.0, is listed below.
2. Is it possible to export the ICD-10 codes to CSV files?  (I set the CSV export to `true` and it only outputs a SNOMED code)

```
 {
    "fullUrl": "urn:uuid:da1119d1-d9eb-69ec-016d-b77ec013ff3a",
    "resource": {
      "resourceType": "Condition",
      "id": "da1119d1-d9eb-69ec-016d-b77ec013ff3a",
      "meta": {
        "profile": [ "http://hl7.org/fhir/us/core/StructureDefinition/us-core-condition-encounter-diagnosis" ]
      },
      "clinicalStatus": {
        "coding": [ {
          "system": "http://terminology.hl7.org/CodeSystem/condition-clinical",
          "code": "resolved"
        } ]
      },
      "verificationStatus": {
        "coding": [ {
          "system": "http://terminology.hl7.org/CodeSystem/condition-ver-status",
          "code": "confirmed"
        } ]
      },
      "category": [ {
        "coding": [ {
          "system": "http://terminology.hl7.org/CodeSystem/condition-category",
          "code": "encounter-diagnosis",
          "display": "Encounter Diagnosis"
        } ]
      } ],
      "code": {
        "coding": [ {
          "system": "http://snomed.info/sct",
          "code": "233678006",
          "display": "Childhood asthma (disorder)"
        }, {
          "system": "http://hl7.org/fhir/sid/icd-10-cm",
          "code": "J45.0",
          "display": "Predominantly allergic asthma"
        } ],
        "text": "Childhood asthma (disorder)"
      },
      "subject": {
        "reference": "urn:uuid:92675303-ca5b-136a-169b-e764c5753f06"
      },
      "encounter": {
        "reference": "urn:uuid:6b561667-bbc7-6b2a-8e5f-e454b66ce83f"
      },
      "onsetDateTime": "2000-05-23T03:37:54-07:00",
      "abatementDateTime": "2017-04-30T03:37:54-07:00",
      "recordedDate": "2000-05-23T03:37:54-07:00"
    },
    "request": {
      "method": "POST",
      "url": "Condition"
    }
  }
```

Thank you, again, for the help.

**Comment by hadleynet** [comment]
Having both SNOMED and ICD-10 in FHIR was the intended behavior - essentially we are treating ICD-10 as a translation while preserving the original.

There's no support for adding ICD-10 codes to the CSV export right now, I guess you could use the mapping file to post-process the CSV and add a new column for ICD codes.

**Reply by telljmc** [reply]
Thank you. I have one last question on the mapping.  Is there a built-in process for mapping SNOMED codes to other code sets besides ICD-10 (eg. CPT) or would those also require post-processing?

**Reply by hadleynet** [reply]
There's nothing built in. If you want to automate it, we've had success using the UMLS [MRCONSO.RRF](https://www.nlm.nih.gov/research/umls/licensedcontent/umlsknowledgesources.html) file to create mappings from one code system to another.

**Comment by telljmc** [comment]
Thank you.
